### DISCUSSION TITLE
Does sampler influence the completeness of a trace?

### DISCUSSION THREAD

**Discussion by lmssy** [open-telemetry/opentelemetry-java-instrumentation#3230] [Q&A]
I set trace sampler rate as 0.1, I store all trace information in ElasticSeach.

I got a trace of my server which spanId is **nveqeCbIYa0=**, parentSpanId is **qzJh8poTULY=**, but I can't find any trace whose spanId is **qzJh8poTULY=**,  it's not reasonable.

Is this because I set the sampler rate not 100%? I think a trace must be complete no matter what sampler rate is.

**Comment by anuraaga** [comment]
Hi @lmssy - as long as you are using a `parentbased_traceidratio` sampler, it will be complete. Is this how you configured it? If so, there is probably some other reason that the span for `qzJh8poTULY` couldn't be exported, perhaps the exporter isn't configured correctly for that server?

**Reply by lmssy** [reply]
hi @anuraaga - Thx for your reply, I check my config is traceidratio, I will change to parentbased_traceidratio and try again. Thanks!

**Reply by lmssy** [reply]
Hi @anuraaga , problem still here, but seems all those situation happen when my server call redis or mysql, is this something related to connection pool or thread pool? I mean the thread of the request to mysql or redis has been changed to another thread of mysql or redis' connection pool, then the trace chain is broken?

**Reply by anuraaga** [reply]
Oh can you confirm, is the trace breaking between two remote servers or is it spans within the same server? Also are you using auto instrumentation Java agent?

**Reply by lmssy** [reply]
Spans within the same server, I use auto instrumentation JavaAgent

**Reply by anuraaga** [reply]
In that case, we'd expect the agent to make sure context is propagated such that the trace is complete. Actually, we wouldn't expect a parent span ID to be set to a non-existent span either way. Can you try using the `logging` exporter to see if the spans are exported correctly when logging? If so, it could be an issue between your collector and Elasticsearch but not the app.
